src/python/pants/reporting/json_reporter.py

```diff
@@ -0,0 +1,141 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import os
+from builtins import list, str
+from collections import defaultdict, namedtuple
+
+from future.moves.itertools import zip_longest
+from future.utils import PY3
+from six import string_types
```

Comment:
We don't want to use this line for two reasons:
1) We should never use Six, as Future is a better and safer version. For example, `six.StringIO` behaves differently between Py2 vs Py3, which is bad. The only exception is for our checker code, as it cannot use Future due to being a universal wheel.
2) We should never use `string_types`, as it contradicts Py3 goal of splitting unicode from bytes. Instead, try to clarify the API, and if we really do need to support both types, use the idiom `isinstance(my_stringy_thing, (str, bytes))`.